lib/yarv.rb

```diff
@@ -118,6 +135,22 @@ def with_frame(iseq)
         @stack = @stack[0..current_stack_length]
       end
     end
+
+    # Pushes a new frame onto the stack, executes the instructions contained
+    # within this instruction sequence, then pops the frame off the stack.
+    def eval(iseq)
```

Comment:
I _think_ eval belongs on the EC because we will need to be able to compile and run new InstructionSequences in the same context.

Reply:
I'm fine with this change moving it to the EC but can you leave the `eval` method here and just make it `eval(context = ExecutionContext.new) = context.eval(self)` to keep that API? That way it still mirrors the `RubyVM::InstructionSequence` one